client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -25,12 +28,16 @@
 
     private final Controller controller;
 
+    private final ScheduledExecutorService executor; 
```

Comment:
It is using the same executor. Look at the constructor. If controller is passed in and hence already has a threadPool we don't create one and this is null.